File: documentation-examples-java/src/java/dev/typr/foundations/docs/oracle/NullableType.java
package dev.typr.foundations.docs.oracle;

import dev.typr.foundations.OracleType;
import dev.typr.foundations.OracleTypes;

import java.util.Optional;

@SuppressWarnings("unused")
public class NullableType {
    //start
    OracleType<Integer> notNull = OracleTypes.numberInt;
    OracleType<Optional<Integer>> nullable = OracleTypes.numberInt.opt();
    //stop
}

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/oracle/NullableType.kt
package dev.typr.foundationskt.docs.oracle

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*

@Suppress("unused")
class NullableType {
    //start
    val notNull: OracleType<Int> = OracleTypes.numberInt
    val nullable: OracleType<Int?> = OracleTypes.numberInt.opt()
    //stop
}
